cirq/ops/fourier_transform.py

```diff
@@ -0,0 +1,187 @@
+    """Phases each computational basis state |k⟩ out of n by e^(i*k/n*exponent).
+    """
+
+    def __init__(self, *, num_qubits: int, exponent: Union[float, sympy.Basic]):
+        self._num_qubits = num_qubits
+        self.exponent = exponent
+
+    def _json_dict_(self):
+        return {
+            'cirq_type': self.__class__.__name__,
+            'num_qubits': self._num_qubits,
+            'exponent': self.exponent
+        }
+
+    def _value_equality_values_(self):
+        return self._num_qubits, self.exponent
+
+    def num_qubits(self) -> int:
+        return self._num_qubits
+
+    def _decompose_(self, qubits):
+        for i, q in enumerate(qubits):
+            yield cirq.Z(q)**(self.exponent / 2**i)
+
+    def _apply_unitary_(self, args: 'cirq.ApplyUnitaryArgs'):
+        if isinstance(self.exponent, sympy.Basic):
+            return NotImplemented
+
+        n = int(np.product([args.target_tensor.shape[k] for k in args.axes]))
+        for i in range(n):
+            p = 1j**(4 * i / n * self.exponent)
+            args.target_tensor[args.subspace_index(big_endian_bits_int=i)] *= p
+
+        return args.target_tensor
+
+    def __pow__(self, power):
+        new_exponent = cirq.mul(self.exponent, power, NotImplemented)
+        if new_exponent is NotImplemented:
+            # coverage: ignore
+            return NotImplemented
+        return PhaseGradientGate(num_qubits=self._num_qubits,
+                                 exponent=new_exponent)
+
+    def _unitary_(self):
+        if isinstance(self.exponent, sympy.Basic):
+            return NotImplemented
+
+        size = 1 << self._num_qubits
+        return np.diag(
+            [1j**(4 * i / size * self.exponent) for i in range(size)])
+
+    def _has_unitary_(self):
+        return not isinstance(self.exponent, sympy.Basic)
+
+    def _is_parameterized_(self):
+        return isinstance(self.exponent, sympy.Basic)
+
+    def _resolve_parameters_(self, resolver):
+        new_exponent = cirq.resolve_parameters(self.exponent, resolver)
+        if new_exponent is self.exponent:
+            return self
+        return PhaseGradientGate(num_qubits=self._num_qubits,
+                                 exponent=new_exponent)
+
+    def __str__(self):
+        return f'Grad[{self._num_qubits}]' + (f'^{self.exponent}'
+                                              if self.exponent != 1 else '')
+
+    def __repr__(self):
+        return 'cirq.PhaseGradientGate(num_qubits={!r}, exponent={})'.format(
+            self._num_qubits, _compat.proper_repr(self.exponent))
+
+    def _circuit_diagram_info_(self, args: 'cirq.CircuitDiagramInfoArgs'):
+        return cirq.CircuitDiagramInfo(
+            wire_symbols=('Grad',) +
+            tuple(f'#{k+1}' for k in range(1, self._num_qubits)),
+            exponent=self.exponent,
+            exponent_qubit_index=0)
+
+
+def QFT(*qubits: 'cirq.Qid', without_reverse: bool = False) -> 'cirq.Operation':
```

Comment:
should this be lowercase?

Reply:
I was trying to make it appear the same as e.g. `cirq.H` and `cirq.SWAP`. A user learning the API won't be aware that it's a function instead of a global constant (and isn't a function technically a global constant anyways?).

Reply:
:+1:

Reply:
@mpharrigan  Any other issue?

Reply:
lgtm